test/framework/cql.go

```diff
@@ -0,0 +1,83 @@
+package framework
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/datastax/go-cassandra-native-protocol/client"
+	"github.com/datastax/go-cassandra-native-protocol/primitive"
+	"github.com/square/certigo/jceks"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// These values are hard-coded in our test fixtures - see test/testdata/fixtures/client-encryption-secret.yaml
+	secretName = "client-encryption-stores"
+	certName   = "cassandra-node_caroot_20220120_134900"
+)
+
+// GetCqlClientConnection establishes a connection to the Stargate ingress's CQL port.
+func (f *E2eFramework) GetCqlClientConnection(
+	ctx context.Context,
+	k8sContext, namespace string,
+	cqlEndpoint HostAndPort,
+	username, password string,
+	ssl bool,
+) (*client.CqlClientConnection, error) {
+
+	var credentials *client.AuthCredentials
+	if username != "" {
+		credentials = &client.AuthCredentials{Username: username, Password: password}
+	}
+	cqlClient := client.NewCqlClient(string(cqlEndpoint), credentials)
+
+	if ssl {
+		secret, err := f.getClientEncryptionSecret(ctx, k8sContext, namespace)
+		if err != nil {
+			return nil, err
+		}
+		rootCas, err := extractCaCertificates(secret)
```

Comment:
Alternatively we could extract the certificate manually once, commit it and just load that file.
Pros:
- this avoids the extra dependency to jceks (keystore management).

Cons:
- we'd have to keep the cert file and config map in sync, but the certs are valid until 2049...
- this is test code and it works so why change it

Reply:
Happy with what we have here, no need to change.